### PR TITLE
[FIX] website: traceback on non html page set as a link

### PR DESCRIPTION
**Pull request by dhp-odoo**
**PURPOSE**
when you add a link in any of the buttons on the website with
non HTML/XML page, it will raise the traceback as we fetch the content to find
id to propose as a list of anchors.

Reproduction Steps:
1) Add a button on the website using the editor.
2) Add link in URL (i.e. /robots.txt)
3) Traceback

**SPECIFICATION**
We are handling the catch if it will raise the traceback.

task-2596392

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JKE-be** [comment]
Hey Dhruv,

Could you add a console.error to keep a trace of the error into the catch()

Thank you

**Comment by dhp-odoo** [comment]
> Hey Dhruv,
> 
> Could you add a console.error to keep a trace of the error into the catch()
> 
> Thank you

Changes Done.

File: addons/website/static/src/js/utils.js [hunk 1]
```diff
@@ -14,15 +14,20 @@ var qweb = core.qweb;
  */
 function loadAnchors(url) {
     return new Promise(function (resolve, reject) {
-        if (url !== window.location.pathname && url[0] !== '#') {
-            $.get(window.location.origin + url).then(resolve, reject);
-        } else {
+        if (url == window.location.pathname || url[0] == '#') {
```
Comment (rdeodoo) [inline addons/website/static/src/js/utils.js:17]:
```suggestion
        if (url === window.location.pathname || url[0] === '#') {
```

File: addons/website/static/src/js/utils.js [hunk 2]
```diff
@@ -14,15 +14,20 @@ var qweb = core.qweb;
...
             resolve(document.body.outerHTML);
+        } else if (url.length && !url.startsWith("http")){
```
Comment (rdeodoo) [inline addons/website/static/src/js/utils.js:19]:
```suggestion
        } else if (url.length && !url.startsWith("http")) {
```

File: addons/website/static/src/js/utils.js [hunk 3]
```diff
@@ -14,15 +14,20 @@ var qweb = core.qweb;
...
+            $.get(window.location.origin + url).then(resolve, reject);
+        } else { // avoid useless query
+            resolve();
         }
     }).then(function (response) {
         return _.map($(response).find('[id][data-anchor=true]'), function (el) {
             return '#' + el.id;
         });
+    }).catch(error => {
+        console.debug(error);
```
Comment (rdeodoo) [inline addons/website/static/src/js/utils.js:29]: not sure this is useful but why not :shrug:

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Just linter error
Working as expected :+1: (good catch for external link)

**Comment by JKE-be** [comment]
@robodoo r+

thank you rdelint for review

**Pull request closed without merging**